src/common/clib-package.c

```diff
@@ -657,6 +657,10 @@ clib_package_new_from_slug_with_package_name(const char *slug, int verbose,
 #ifdef HAVE_PTHREADS
       init_curl_share();
       _debug("GET %s", json_url);
+      if (res) {
+        // clean up when retrying
+        http_get_free(res);
+      }
```

Comment:
IIRC, `http_get_free()` has a NULL check. We can _always_ do this:

```suggestion
      // clean up when retrying
      http_get_free(res);
```